server/database.sql

```diff
@@ -1,150 +1,378 @@
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `notifications` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `heading` varchar(110) NOT NULL,
+  `description` varchar(255) DEFAULT NULL,
+  `created_at` datetime DEFAULT CURRENT_TIMESTAMP,
+  `creator` varchar(45) NOT NULL,
   PRIMARY KEY (`id`),
   KEY `notifications_fk_creator` (`creator`),
-  CONSTRAINT `notifications_fk_creator` 
-    FOREIGN KEY (`creator`) 
-    REFERENCES `users` (`username`) 
-    ON DELETE CASCADE 
-    ON UPDATE CASCADE) 
-  ENGINE = InnoDB 
-  DEFAULT CHARACTER SET = utf8mb4 
-  COLLATE = utf8mb4_0900_ai_ci;
-
-
--- -----------------------------------------------------
--- Table `user_groups`
--- -----------------------------------------------------
-DROP TABLE IF EXISTS `user_groups` ;
-
-CREATE TABLE IF NOT EXISTS `user_groups` (
-  `id` INT NOT NULL AUTO_INCREMENT,
-  `username` VARCHAR(45) NOT NULL,
-  `group_id` INT NOT NULL,
-  `is_group_moderator` TINYINT NOT NULL,
+  CONSTRAINT `notifications_fk_creator` FOREIGN KEY (`creator`) REFERENCES `users` (`username`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `questions`
+--
+
+DROP TABLE IF EXISTS `questions`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `questions` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `creator` varchar(45) NOT NULL,
+  `type` enum('mcq','subjective') NOT NULL,
+  `name` varchar(100) NOT NULL,
+  `problem_statement` text NOT NULL,
+  `input_format` text,
+  `output_format` text,
+  `constraints` text,
+  `options` text,
+  `correct` int DEFAULT NULL,
+  `difficulty` enum('easy','medium','hard') DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_questions_creator_idx` (`creator`),
+  CONSTRAINT `fk_questions_creator` FOREIGN KEY (`creator`) REFERENCES `users` (`username`) ON UPDATE RESTRICT
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `questions_editors`
+--
+
+DROP TABLE IF EXISTS `questions_editors`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `questions_editors` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `question_id` int NOT NULL,
+  `editor` varchar(45) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_questions_editors_1_idx` (`question_id`),
+  KEY `fk_questions_editors_2_idx` (`editor`),
+  CONSTRAINT `fk_questions_editors_1` FOREIGN KEY (`question_id`) REFERENCES `questions` (`id`) ON DELETE CASCADE ON UPDATE RESTRICT,
+  CONSTRAINT `fk_questions_editors_2` FOREIGN KEY (`editor`) REFERENCES `users` (`username`) ON DELETE RESTRICT ON UPDATE CASCADE
```

Comment:
Similarly, 1 more constraint needs to be added to avoid unnecessary duplicate entries -
```
UNIQUE KEY `unique_questions_editors` (`question_id`,`editor`)
```